pkg/controller/roleset/rolesyncer.go

```diff
@@ -438,20 +442,24 @@ func (s *StatelessRoleSyncer) Rollout(ctx context.Context, roleSet *orchestratio
 	}
 	activePods, _ := filterActivePods(allPods)
 	expectedReplicas := getRoleReplicas(role)
-	updated, outdated := filterUpdatedPods(activePods, ctrlutil.ComputeHash(&role.Template, nil))
-	klog.Infof("[StatelessRoleSyncer.Rollout] roleset %s/%s role %s updated %d, outdated %d, expectedReplicas %d, hash %s", roleSet.Namespace, roleSet.Name, role.Name, len(updated), len(outdated), expectedReplicas, ctrlutil.ComputeHash(&role.Template, nil))
+	updated, outdated := filterUpdatedPods(activePods, s.computeHashFunc(&role.Template, nil))
+	klog.Infof("[StatelessRoleSyncer.Rollout] roleset %s/%s role %s updated %d, outdated %d, expectedReplicas %d, hash %s", roleSet.Namespace, roleSet.Name, role.Name, len(updated), len(outdated), expectedReplicas, s.computeHashFunc(&role.Template, nil))
 	ready, _ := filterReadyPods(activePods)
 	deleteBudget := int32(len(ready)) - expectedReplicas + MaxUnavailable(role)
 	sortPodsByActive(outdated)
 	// 1. delete outdated pods
 	for i := 0; i < len(outdated); i++ {
-		if podutil.IsPodReady(outdated[i]) {
+		// If the pod is not ready and outdated, we can always delete it.
+		if !podutil.IsPodReady(outdated[i]) {
+			toDelete = append(toDelete, outdated[i])
+		} else {
```

Comment:
this is a logic change.. I think it's better to have a code improvement for fix separately. After the change, all unready pods will be processed.